From adbeb4ba62a4afcea8f385f82482dcb0a710680d Mon Sep 17 00:00:00 2001
From: uti <uti.edun@chainguard.dev>
Date: Sat, 22 Feb 2025 19:52:24 +0000
Subject: [PATCH] fix: cast result to intptr_t in
 vmprof_write_header_for_jit_addr calls and ensure PATH_MAX is defined

---
 rpython/rlib/rvmprof/src/shared/vmp_stack.c | 8 ++++----
 rpython/translator/c/src/signals.h          | 5 +++--
 2 files changed, 7 insertions(+), 6 deletions(-)

diff --git a/rpython/rlib/rvmprof/src/shared/vmp_stack.c b/rpython/rlib/rvmprof/src/shared/vmp_stack.c
index 28041d33f0..c5533a0c75 100644
--- a/rpython/rlib/rvmprof/src/shared/vmp_stack.c
+++ b/rpython/rlib/rvmprof/src/shared/vmp_stack.c
@@ -116,7 +116,7 @@ int vmp_profiles_python_lines(void) {
 #ifdef PYPY_JIT_CODEMAP
     else if (frame->kind == VMPROF_JITTED_TAG) {
         intptr_t pc = ((intptr_t*)(frame->value - sizeof(intptr_t)))[0];
-        *depth = vmprof_write_header_for_jit_addr(result, *depth, pc, max_depth);
+        *depth = vmprof_write_header_for_jit_addr((intptr_t*)result, *depth, pc, max_depth);
     }
 #endif
 
@@ -158,7 +158,7 @@ static PY_STACK_FRAME_T * _write_python_stack_entry(PY_STACK_FRAME_T * frame, vo
 #ifdef PYPY_JIT_CODEMAP
     else if (frame->kind == VMPROF_JITTED_TAG) {
         intptr_t pc = ((intptr_t*)(frame->value - sizeof(intptr_t)))[0];
-        *depth = vmprof_write_header_for_jit_addr(result, *depth, pc, max_depth);
+        *depth = vmprof_write_header_for_jit_addr((intptr_t*)result, *depth, pc, max_depth);
     }
 #endif
 
@@ -343,7 +343,7 @@ int vmp_walk_and_record_stack(_PyInterpreterFrame *frame, void ** result,
             return vmp_walk_and_record_python_stack_only(frame, result, max_depth, depth, pc);
 #ifdef PYPY_JIT_CODEMAP
         } else if (pypy_find_codemap_at_addr(rip, &start_addr) != NULL) {
-            depth = vmprof_write_header_for_jit_addr(result, depth, pc, max_depth);
+            depth = vmprof_write_header_for_jit_addr((intptr_t*)result, depth, pc, max_depth);
             return vmp_walk_and_record_python_stack_only(frame, result, max_depth, depth, pc);
 #endif
         } else {
@@ -497,7 +497,7 @@ int vmp_walk_and_record_stack(PY_STACK_FRAME_T *frame, void ** result,
             return vmp_walk_and_record_python_stack_only(frame, result, max_depth, depth, pc);
 #ifdef PYPY_JIT_CODEMAP
         } else if (pypy_find_codemap_at_addr(rip, &start_addr) != NULL) {
-            depth = vmprof_write_header_for_jit_addr(result, depth, pc, max_depth);
+            depth = vmprof_write_header_for_jit_addr((intptr_t*)result, depth, pc, max_depth);
             return vmp_walk_and_record_python_stack_only(frame, result, max_depth, depth, pc);
 #endif
         } else {
diff --git a/rpython/translator/c/src/signals.h b/rpython/translator/c/src/signals.h
index 63715f4f1e..25da8080f7 100644
--- a/rpython/translator/c/src/signals.h
+++ b/rpython/translator/c/src/signals.h
@@ -23,8 +23,9 @@ int pypysig_poll(void);   /* => signum or -1 */
 RPY_EXTERN
 void pypysig_pushback(int signum);
 
-#define PATH_MAX 1024
-
+#ifndef PATH_MAX
+ #define PATH_MAX 1024
+#endif
 /* When a signal is received, pypysig_counter is set to -1. */
 struct pypysig_long_struct_inner {
     Signed value;
-- 
2.48.1

